cmd/package_managers.go

```diff
@@ -108,6 +113,51 @@ func fetchGitRepositoryFromNPM(packageName string, packageManager pmc.Client) (s
 	return v.Objects[0].Package.Links.Repository, nil
 }
 
+func repoIfValid(url string) string {
+	match := _GITHUB_DOMAIN_REGEXP.FindStringSubmatch(url)
+	if len(match) >= 3 {
+		return fmt.Sprintf("https://github.com/%s/%s", match[1], match[2])
+	}
+
+	match = _GITHUB_SUBDOMAIN_REGEXP.FindStringSubmatch(url)
+	if len(match) >= 3 {
+		return fmt.Sprintf("https://github.com/%s/%s", match[1], match[2])
+	}
+
+	match = _GITLAB_DOMAIN_REGEXP.FindStringSubmatch(url)
+	if len(match) >= 3 {
+		return fmt.Sprintf("https://gitlab.com/%s/%s", match[1], match[2])
+	}
+	return ""
+}
+
+func findGitRepositoryInPYPIResponse(packageName string, response io.Reader) (string, error) {
+	v := &pypiSearchResults{}
+	err := json.NewDecoder(response).Decode(v)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("failed to parse pypi package json: %v", err))
+	}
+
+	v.Info.ProjectURLs["key_not_used"] = v.Info.ProjectURL
+	validURLs := make(map[string]any)
+	for _, url := range v.Info.ProjectURLs {
+		if repo := repoIfValid(url); repo != "" {
+			validURLs[repo] = nil
+		}
+	}
+
+	if len(validURLs) > 1 {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("found too many possible source repos for pypi package: %s", packageName))
+	}
+
+	for url, _ := range validURLs {
+		return url, nil
+	}
```

Comment:
If I understand this code correctly, the `validURLs` map functions as a set to de-duplicate repo URLs. However, it doesn't appear that the full set contents ever matter since `len(validURLs) > 1` is an error case. Is the set even really necessary, then? Could you just have a single string variable and bail out the first time you find a URL that does not match the one you've already seen?

Reply:
Good call! Done